src/printer.js

```diff
@@ -1720,7 +1720,8 @@ function printArgumentsList(path, options, print) {
     lastArg.type === "FunctionExpression" ||
     lastArg.type === "ArrowFunctionExpression" &&
       (lastArg.body.type === "BlockStatement" ||
-        lastArg.body.type === "ArrowFunctionExpression") ||
+        lastArg.body.type === "ArrowFunctionExpression" ||
```

Comment:
I worked through this on my livestream, and had the exact same question! I removed the `body.type` check and saw why from the tests. I can't remember exactly why, but try it out.

I think the part of the reason is https://github.com/jlongster/prettier/issues/61 actually. We allowed block-less arrow functions to break on the content, so if you have `x => this.foo().bar()`, it could actually break right after the arrow. That would look weird though, or even totally broken if there is more complex content, if we allow this last arg grouping behavior. So we only break blocks.

However, turns out of course there are other things we want to allow to break: objects, arrays, etc. But not everything. Feel free to think about this and let me know if there's a better solution! I'm fine if there's a different way that works for everything.